std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(is(T : const char[]) || is(T : const wchar[]) || is(T : const dchar[]))
         {
+            import std.utf : byUTF;
             type_tag = JSON_TYPE.STRING;
-            store.str = arg.idup;
+            store.str = byUTF!char(arg).array.idup;
```

Comment:
I think that just `store.str = arg.byUTF!char.array;` should be enough. It should be inferred automatically as `string` because mutable result of `pure` functions is automatically convertible to immutable.